sourced/ml/core/algorithms/token_parser.py

```diff
@@ -166,6 +186,36 @@ def ret(name):
             if last:
                 yield from ret(last)
 
+    def __pre_split_token(self, token: str) -> [str]:
+        """Split a token by non-alphanumeric characters"""
+        splits = re.split('[^a-zA-Z0-9]', token)
```

Comment:
We should use `NAME_BREAKUP_RE`. This additionally ignores digits. We do not handle them in the nn splitter anyway.